projects/core/src/occ/adapters/product/occ-product.adapter.ts

```diff
@@ -26,6 +26,35 @@ export class OccProductAdapter implements ProductAdapter {
     protected requestsOptimizer: OccRequestsOptimizerService
   ) {}
 
+  loadRealTimeStock(
+    productCode: string,
+    unit: string
+  ): Observable<ProductAvailabilities> {
+    const availabilityUrl = this.occEndpoints.buildUrl(
+      'productAvailabilities',
+      {
+        urlParams: {
+          productCode: productCode,
+          sapCode: unit,
+        },
+      }
+    );
+
+    return this.http.get(availabilityUrl).pipe(
+      take(1),
+      distinctUntilChanged(),
+      switchMap((availabilities: any) => {
+        const quantity =
```

Comment:
MINOR: can we just do
`return of(availabilities?.availabilityItems?.[0]?.unitAvailabilities?.[0] || {});`

Reply:
done